src/compositemark/callout.ts

```diff
@@ -0,0 +1,131 @@
+import {Config} from '../config';
+import {Encoding} from '../encoding';
+import * as log from '../log';
+import {MarkConfig, MarkDef} from '../mark';
+import {GenericUnitSpec, LayerSpec} from '../spec';
+
+
+export const CALLOUT: 'callout' = 'callout';
+export type CALLOUT = typeof CALLOUT;
+
+export interface CalloutDef {
+  type: CALLOUT;
+  calloutAngle?: number;
+  calloutOffset?: number;
+  calloutLength?: number;
+  labelOffset?: number;
+  callout?: MarkDef;
```

Comment:
MarkDef is a bit too much?   You don't want them to change the mark type of the `callout`.